pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -382,6 +449,83 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 	return nil
 }
 
+func getContainerWithName(pod *v1.Pod, name string) *v1.Container {
+	for idx, container := range pod.Spec.Containers {
+		if strings.Compare(container.Name, name) == 0 {
+			return &pod.Spec.Containers[idx]
+		}
+	}
+	return nil
+}
+
+// Add an environment variable with the given value to the environments
+// variables of the given container, potentially replacing an env var that already exists
+// with this name
+func addOrReplaceEnv(container *v1.Container, envKey string, envValue string) {
+	if container.Env == nil {
+		container.Env = []v1.EnvVar{}
+	}
+
+	for i, envVar := range container.Env {
+		if envVar.Name == envKey {
+			container.Env[i].Value = envValue
+			return
+		}
+	}
+
+	container.Env = append(container.Env, v1.EnvVar{
+		Name:  envKey,
+		Value: envValue,
+	})
+}
+
+func createModelContainer(image string, modelPath string) *v1.Container {
+	modelContainer := &v1.Container{
+		Name:  ModelcarContainerName,
+		Image: image,
+		VolumeMounts: []v1.VolumeMount{
+			{
+				Name:      StorageInitializerVolumeName,
+				MountPath: getParentDirectory(modelPath),
+				ReadOnly:  false,
+			},
+		},
+		Args: []string{
+			"sh",
+			"-c",
+			// $$$$ gets escaped by YAML to $$, which is the current PID
+			fmt.Sprintf("ln -s /proc/$$$$/root/models %s && tail -f /dev/null", modelPath),
```

Comment:
If we're splitting hairs `tail` will invoke `read` periodically (every second IIRC), this doesn't impact performance

```suggestion
			fmt.Sprintf("ln -s /proc/$$$$/root/models %s && sleep infinity", modelPath),
```

Reply:
The reason for  using `tail` over `sleep` was mostly portability, also to keep the requirements for the base image small. 

From https://stackoverflow.com/a/45396600:

> Portability
> This is an important and valid concern:
> sleep infinity is a GNU coreutils extension not contemplated in 
 POSIX. GNU's implementation also supports a "fancy" syntax for time durations, like sleep 1h 5.2s while POSIX only allows a positive integer (e.g. sleep 0.5 is not allowed).
> Some compatible implementations: GNU coreutils, FreeBSD (at least from version 8.2?), Busybox (requires to be compiled with options FANCY_SLEEP and FLOAT_DURATION).
> The strtod behaviour is C and POSIX compatible (i.e. strtod("infinity", 0) is always valid in C99-conformant implementations, see §7.20.1.3).

Said that, probably the final solution should add its own command that does the linking & waiting (and signal handling) on its own (idealy leveraging the `pause()` system call), and then provide the base image from kserve.